vajram/vajram-krystex/src/main/java/com/flipkart/krystal/vajramexecutor/krystex/SessionInputDecorator.java

```diff
@@ -0,0 +1,118 @@
+package com.flipkart.krystal.vajramexecutor.krystex;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+import com.flipkart.krystal.data.InputValue;
+import com.flipkart.krystal.data.Inputs;
+import com.flipkart.krystal.data.ValueOrError;
+import com.flipkart.krystal.datatypes.JavaDataType;
+import com.flipkart.krystal.krystex.MainLogic;
+import com.flipkart.krystal.krystex.MainLogicDefinition;
+import com.flipkart.krystal.krystex.decoration.MainLogicDecorator;
+import com.flipkart.krystal.vajram.Vajram;
+import com.flipkart.krystal.vajram.adaptors.DependencyInjectionAdaptor;
+import com.flipkart.krystal.vajram.inputs.Input;
+import com.flipkart.krystal.vajram.inputs.InputSource;
+import com.flipkart.krystal.vajram.inputs.VajramInputDefinition;
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public record SessionInputDecorator(
+    Vajram<?> vajram, DependencyInjectionAdaptor<?> dependencyInjectionAdaptor)
```

Comment:
Since we are writing custom adaptors, we can accept the guice Injector directly instead. and we can name this class GuiceInjectionDecorator, that way we don't need this interface

Reply:
I was thinking the service which will use Krsytal will implement their own injector (custom or readily available). Also, if we directly keep GuiceInjectionDecorator in Vajram module, then it will add the guice dependency in Vajram module which we don't want to do, right?